### PR TITLE
Local Development + extra exports

### PR DESCRIPTION
**Pull request by jburrow**
The context to the PR is i am working on a time-series plugin which can render cpu, memory in addition to the other plugins you have. I am also want to render multiple flame-plugins are the same time for comparisions use-case. 

This PR is a small one that
 - Introduces a simple local-dev experience with autoreload from npm start
 - Exports the {class} + default export to make imports in typescript "nicer" import {classname} from "script"
 - Export all the types.ts - some were missing.
 - Ability to override plugin instance name to support multiple instances of same pluging (for comparision) i have additional changes locally.

Thanks for your time and consideration.

James

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +42,7 @@
     "@types/color": "^3.0.3",
     "@types/events": "^3.0.0",
     "@typescript-eslint/eslint-plugin": "^5.39.0",
-    "@typescript-eslint/parser": "^5.39.0",
+    "@typescript-eslint/parser": "^5.39.0",
```
Comment (Smrtnyk) [inline package.json:45]: this whitespaces got added?

File: rollup.config.js [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
+import babel from 'rollup-plugin-babel';
+import builtins from 'rollup-plugin-node-builtins';
+import resolve from 'rollup-plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import json from '@rollup/plugin-json';
+import html from '@rollup/plugin-html';
+import cleaner from 'rollup-plugin-cleaner';
+import { template } from './example/src/template.js';
+import typescript from '@rollup/plugin-typescript';
+
```
Comment (Smrtnyk) [inline rollup.config.js:10]: this whitespace seems superfluous

File: src/flame-chart.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import MarksPlugin from './plugins/marks-plugin';
 import { Colors, Data, Marks, Waterfall } from './types';
 import UIPlugin from './plugins/ui-plugin';
 
+
```
Comment (Smrtnyk) [inline src/flame-chart.ts:11]: same here

File: src/flame-chart.ts [hunk 4]
```diff
@@ -34,7 +35,7 @@ export type FlameChartOptions = {
 
 const defaultSettings: FlameChartSettings = {};
 
-export default class FlameChart extends FlameChartContainer<FlameChartStyles> {
+export class FlameChart extends FlameChartContainer<FlameChartStyles> {
```
Comment (Smrtnyk) [inline src/flame-chart.ts:38]: named export and default export of the same thing?
not sure how to feel about that

**Review by Smrtnyk — COMMENTED** [review]
I am not an owner of this repo
just a contributor with rights to merge stuff

A proper review can give you Nikolay

From my side, I noticed there is some unneeded noise with extra added lines in files
PR would be smaller without those

File: src/flame-chart.ts (hunk 4 above)
Comment (jburrow) [reply src/flame-chart.ts:38]: I can understand how it looks odd - but it makes for a nicer typescript api (IMHO). It is backwards compat, non-breaking.

I have a lot more chagnes pending - converting the example to typescript - and the code looks cleaner without using the default export.

File: package.json (hunk 1 above)
Comment (jburrow) [reply package.json:45]: agreed - sorry + fixed

File: rollup.config.js (hunk 2 above)
Comment (jburrow) [reply rollup.config.js:10]: agreed - sorry + fixed

File: src/flame-chart.ts (hunk 3 above)
Comment (jburrow) [reply src/flame-chart.ts:11]: agreed - sorry + fixed

File: rollup.config.js (hunk 2 above)
Comment (jburrow) [reply rollup.config.js:10]: This is just ";" missing from end of lines.

image

Comment (jburrow) [reply rollup.config.js]: removed from PR

**Comment by jburrow** [comment]
Do you need me to do anything else?

**Review by pyatyispyatil — CHANGES_REQUESTED** [review]
It would be very nice to have named exports for all plugins. Otherwise it will be very inconvenient for our users.

**Comment by jburrow** [comment]
I have updated all the plugins to have have named exports, and updated their usage in index.ts to use the named instances.
This change is non-breaking.

I am sorry to say i don't know how to resolve this conflicts with package.json

**Review by pyatyispyatil — APPROVED** [review]

**Comment by jburrow** [comment]
many thanks

**Pull request merged**